File: scala/pascals-triangle/src/main/scala/PascalsTriangle.scala
object PascalsTriangle{
  def rows(numberOfRows: Int): List[List[Int]] = {
    if(numberOfRows <= 0) Nil
    else List.iterate(List(1), numberOfRows) {prev => 0 +: prev zip prev :+ 0 map Function.tupled(_ + _)}
  }
}

File: elixir/pascals-triangle/lib/pascals_triangle.ex
defmodule PascalsTriangle do
  @doc """
  Calculates the rows of a pascal triangle
  with the given height
  """
  @spec rows(integer) :: [[integer]]
  def rows(num) do
    Stream.iterate([1], fn prev -> 
      [1 | Enum.map(Enum.chunk_every(prev, 2, 1), &Enum.sum/1)]
    end)
    |> Enum.take(num)
  end
end
